lib/core/src/sdk.rs

```diff
@@ -56,7 +56,50 @@ pub struct LiquidSdk {
 
 impl LiquidSdk {
     pub async fn connect(req: ConnectRequest) -> Result<Arc<LiquidSdk>> {
-        let config = req.config;
+        // If we can fetch a proxy swapper URL, extract referral ID and dynamically set it in the config
+        let (boltz_api_base_url, referral_id) = match &req.config.network {
```

Comment:
Also perhaps we can persist the last swapper url so in case we get error from the service we can use the persisted value.

Reply:
> I think it is better to pass the url directly to the swapper instance and let it parse the query instead of doing it here.

Done: 56e0942ee0737362cc10055468f91f24155cc95b

Reply:
> Also perhaps we can persist the last swapper url so in case we get error from the service we can use the persisted value.

Done: 31a037cc9c7fcfc8eeed2c07b2f6c3a92d5a07fd